erpnext/agriculture/doctype/plant_batch/plant_batch.py

```diff
@@ -115,4 +115,30 @@ def make_disease_diagnosis(source_name, target_doc=None):
 			}
 		}}, target_doc)
 
-	return target_doc
\ No newline at end of file
+	return target_doc
+
+@frappe.whitelist()
+def destroy_plant_batch(destroy_count, reason_note, reference_name=None):
+	plant_batch = frappe.get_doc("Plant Batch",reference_name)
+	if plant_batch.untracked_count == 0 :
+		frappe.throw(_("Cannot Destroy Plant Batch as there is no untracked count"))
+
+	if int(destroy_count) < 0 :
+		frappe.throw(_("The Destroy Count ({0}) should be more than 0").format(destroy_count))
+
+	if plant_batch.untracked_count < int(destroy_count):
+		frappe.throw(_("The Destroy Count ({0}) should be less or equal to the untracked count ({1})").format(destroy_count,plant_batch.untracked_count))
+
+	destroy_plant_batch = frappe.get_doc(
+		dict(doctype='Destroy Plant Batch',
+			plant_batch = reference_name,
+			destroy_count = destroy_count,
+			reason_note = reason_note,
+			actual_date = getdate(nowdate())
+			)
+			).insert()
+	destroy_plant_batch.submit()
```

Comment:
- Submit this document after updating the values of `untracked_count` and `destroyed_count` after inserting it.
- Now there is one unnecessary update being done in db.